docs/api/target.rst

```diff
@@ -118,52 +118,88 @@ blt_add_library
+SOURCES
+  List of all sources to be added
 
-Note: Object libraries do not follow CMake's transitivity rules until 3.12.
-BLT will add the various information provided in this macro and its
-dependencies in the order you provide them to help.
+HEADERS
+  List of all headers to be added
 
-The INCLUDES argument allows you to define what include directories are
-needed by any target that is dependent on this library.  These will
-be inherited by CMake's target dependency rules.
+INCLUDES
+  List of include directories both used by this target and inherited by dependent
+  targets
 
-The DEFINES argument allows you to add needed compiler definitions that are
-needed by any target that is dependent on this library.  These will
-be inherited by CMake's target dependency rules.
+DEFINES
+  List of compiler defines both used by this target and inherited by dependent
+  targets
 
-If given a DEPENDS_ON argument, it will add the necessary includes and 
-libraries if they are already registered with blt_register_library.  If 
-not it will add them as a CMake target dependency.
+DEPENDS_ON
+  List of CMake targets and BLT registered libraries that this library
+  depends on
 
-In addition, this macro will add the associated dependencies to the given
-library target. Specifically, it will add the dependency for the CMake target
-and for copying the headers for that target as well.
+OUTPUT_NAME
+  Override built file name of library (defaults to <NAME>)  
 
-The OUTPUT_DIR is used to control the build output directory of this 
-library. This is used to overwrite the default lib directory.
-OUTPUT_NAME is the name of the output file; the default is NAME.
+OUTPUT_DIR
+  Directory that this target will built to
 
-It's useful when multiple libraries with the same name need to be created
-by different targets. NAME is the target name, OUTPUT_NAME is the library name.
-CLEAR_PREFIX allows you to remove the automatically appended "lib" prefix
-from your built library.  The created library will be foo.a instead of libfoo.a.
-FOLDER is an optional keyword to organize the target into a folder in an IDE.
+SHARED
+  Builds library as shared and overrides global BUILD_SHARED_LIBS (defaults to OFF)
+
+OBJECT
+  Create an Object library
+
+CLEAR_PREFIX
+  Removes library prefix (defaults to 'lib' on linux)
+
+FOLDER
+  Name of the IDE folder to ease organization
 
-This is available when ENABLE_FOLDERS is ON and when the cmake generator
-supports this feature and will otherwise be ignored. 
+This macro supports three types of libraries automatically: normal, header-only,
+or object.
 
-Note: Do not use with header-only (INTERFACE) libraries, as this will generate 
-a CMake configuration error.
+Normal libraries are libraries that have sources that are compiled and linked into a single
+library and have headers that go along with them (unless it's a Fortran library).
+
+Header-only libraries are useful when you do not the library separately compiled or 
```

Comment:
Please rephrase this sentence and add a missing word:

> when you do not <> the library 

Perhaps:
> when you do not **need** ?

Reply:
More generally, "Header-only" libraries are for libraries that do not have source files.